packages/ra-core/src/form/Form.tsx

```diff
@@ -42,7 +46,9 @@ import { useAugmentedForm } from './useAugmentedForm';
  *
  * @link https://react-hook-form.com/docs/useformcontext
  */
-export const Form = (props: FormProps) => {
+export const Form = <RecordType extends RaRecord = any>(
```

Comment:
Form can be used in a Create page, where there is no `id`. Therefore it's not necessarily operating on a `RaRecord`

Reply:
Bump